### PR TITLE
Improve rate limit policy

### PR DESCRIPTION
**Pull request by y-tabata**
This PR improves the rate limit policy according to the comments of [PR #665](https://github.com/3scale/apicast/pull/665) and addresses to [Issue #667](https://github.com/3scale/apicast/issues/667).

### CODE REVIEW

File: gateway/src/apicast/policy/rate_limit/rate_limit.lua [hunk 1]
```diff
@@ -10,6 +10,10 @@ local limit_traffic = require "resty.limit.traffic"
 local ts = require ('apicast.threescale_utils')
 local tonumber = tonumber
 local next = next
+local format = string.format
+local insert = table.insert
+local os_date = os.date
```
Comment (mikz) [inline gateway/src/apicast/policy/rate_limit/rate_limit.lua]: Better to use `ngx.time` or `ngx.now` as they are way more performant in scope of OpenResty.

File: spec/policy/rate_limit/rate_limit_spec.lua [hunk 2]
```diff
@@ -234,6 +234,7 @@ describe('Rate limit policy', function()
       assert.equal('10', window)
 
       os.execute("sleep 1")
```
Comment (mikz) [inline spec/policy/rate_limit/rate_limit_spec.lua]: Haven't noticed this earlier. There is `ngx.sleep()` that is better than spawning a subprocess.

**Comment by y-tabata** [comment]
@mikz The `build` error and `s2i-builder` error have nothing to do with this PR, right?

**Comment by davidor** [comment]
@y-tabata The error has nothing to do with your PR, sorry.

**Comment by y-tabata** [comment]
@mikz @davidor Is there anything I can do to help move this PR forward?

**Comment by mikz** [comment]
@y-tabata I'm sorry, but the recent changes introduced in https://github.com/3scale/apicast/pull/696 conflict with this PR.
Could you resolve those conflicts?

**Comment by y-tabata** [comment]
@mikz Sure, no problem.

**Comment by y-tabata** [comment]
move to #703

**Pull request closed without merging**